### ISSUE TITLE
Refactor conditional multitask Lightning flow to use split-specific train and QuerySeq eval datasets

### ISSUE THREAD

**Issue by gkondas** [mmcdermott/EveryQuery-conditional#30]
## Goal

Refactor the conditional multitask training/evaluation plumbing so that Lightning owns both execution paths:

- **training + validation:** `MultitaskBoundaryPytorchDataset` → `MultitaskBoundaryBatch` → dense all-vocabulary `(B, K, V)` logits/loss;
- **test + prediction:** `QuerySeqMultitaskEvalDataset` → `MultitaskEvalBatch` → one scalar logit/probability per row, shape `(B,)`.

Work from the tip of `feat/queryseq-eval-consolidation`.

## Why

The branch already has the correct model-level separation:

- `ConditionalMultitaskARModel.forward(batch)` is the training pass. It projects every real window onto the whole vocabulary, requires dense `targets`, and computes the masked BCE loss.
- `ConditionalMultitaskARModel.score_final_query(batch, scored_codes)` is the QuerySeq evaluation pass. It scores only the last real window and that row's final query code, returning `(B,)` without materializing `(B, K, V)` or reading `targets`.
- `QuerySeqMultitaskEvalDataset` already adapts a `QuerySeqSchema` row into teacher-forced earlier queries plus one final scored query.

However, the surrounding Lightning plumbing does not express that separation:

- `ResumableDatamodule` currently uses one `data_class` for train, validation, and test.
- `ConditionalMultitaskLightningModule.predict_step` currently expects `MultitaskBoundaryBatch` and calls the dense training forward.
- `predict_multitask.py` manually builds its dataset/dataloader, moves batches to the device, and runs inference outside `Trainer.predict`.

## Required loop contracts

| Lightning loop | Dataset | Batch | Model path |
|---|---|---|---|
| `training_step` | `MultitaskBoundaryPytorchDataset(train)` | `MultitaskBoundaryBatch` | dense `forward`, `(B,K,V)` |
| `validation_step` | `MultitaskBoundaryPytorchDataset(tuning)` | `MultitaskBoundaryBatch` | dense `forward`, `(B,K,V)` |
| `test_step` | `QuerySeqMultitaskEvalDataset(held_out)` | `MultitaskEvalBatch` | `score_final_query`, `(B,)` |
| `predict_step` | `QuerySeqMultitaskEvalDataset(configured split)` | `MultitaskEvalBatch` | `score_final_query`, `(B,)` |

**Keep validation training-style.** Checkpointing and early stopping currently monitor `tuning/loss`, which is the dense multitask objective; this issue must not silently change its meaning.

### 1. Add a conditional-multitask datamodule

Add a focused datamodule (for example, `ConditionalMultitaskDataModule`) rather than teaching the generic `ResumableDatamodule` about this architecture-specific dataset pairing.

It should:

- preserve the existing `StatefulDataLoader` behavior for the training loader;
- construct `MultitaskBoundaryPytorchDataset` for `train_dataset` and `val_dataset`;
- construct `QuerySeqMultitaskEvalDataset` for `test_dataset` and `predict_dataset`;
- use non-shuffled/sequential test and prediction dataloaders;
- share the checkpoint's cohort/sequence settings across both dataset types;
- allow the training multitask-label root and QuerySeq evaluation-grid root to differ;
- replace only `task_labels_dir` when deriving the evaluation data config, matching the behavior of the existing `build_eval_dataset()`;
- pass the evaluation adapter the checkpoint-compatible values for:
  - `strip_delta_tokens`;
  - `expected_vocab_size`;
  - `max_windows`.

The QuerySeq grid root must be optional for ordinary `trainer.fit()`. Training should not require an evaluation artifact. Raise a clear error only when a test/predict dataset or dataloader is requested without an evaluation root.

### 2. Make the LightningModule loop-specific

In `ConditionalMultitaskLightningModule`:

- keep `training_step` dense over `MultitaskBoundaryBatch`;
- explicitly keep/override `validation_step` as the same dense objective;
- override `test_step` for `MultitaskEvalBatch`;
- replace the existing dense `predict_step` with a scalar QuerySeq version.

Factor the shared scalar operation if useful:

```python
logits = self.model.score_final_query(batch, batch.scored_codes)  # (B,)
```

`test_step` should compute/log an appropriate scalar held-out loss from `batch.labels`. `predict_step` should return, at minimum:

```python
{
    "probs": torch.sigmoid(logits),  # (B,)
    "labels": batch.labels,          # (B,)
    "scored_codes": batch.scored_codes,  # (B,)
}
```

Do not select behavior using `self.training`: validation, test, and prediction are all eval-mode loops but require different contracts here.

### 3. Move `EQ_predict_multitask` onto `Trainer.predict`

Refactor `src/every_query/predict/predict_multitask.py` to invoke the Lightning prediction loop using the new datamodule/dataloader contract.

Remove the manual device-transfer/inference plumbing that Lightning now owns (notably `_to_device` and the manual model iteration in `run_inference`). Preserve:

- the current CLI inputs;
- output columns and semantics;
- final-query labels;
- grid-to-output row alignment checks;
- the guarantee that dense `(B,K,V)` tensors are never constructed during prediction.

Use an inference-appropriate Trainer configuration: no training logger/checkpointing side effects.

Initially require or enforce single-device prediction unless row identity is carried through the batch. Lightning/distributed sampling must not silently reorder outputs relative to `dataset.schema_df`. If multi-device prediction is supported, add an explicit source-row index to the batch/output and restore order before `predictions_to_df`.

### 4. Configuration and compatibility

Update the relevant Hydra configs so the two independent label roots are unambiguous:

- training multitask labels/packed sidecars;
- QuerySeq evaluation-grid `eval/` root.

Keep existing checkpoints usable. Older `resolved_config.yaml` files name `ResumableDatamodule`; `EQ_predict_multitask` must continue to load their cohort/sequence settings and construct the new prediction datamodule rather than requiring the historical datamodule itself to provide `predict_dataloader()`.

Do not change the model parameterization or checkpoint state dict as part of this issue.

## Tests / acceptance criteria

- [ ] `train_dataloader()` yields `MultitaskBoundaryBatch`.
- [ ] `val_dataloader()` yields `MultitaskBoundaryBatch`.
- [ ] `test_dataloader()` and `predict_dataloader()` yield `MultitaskEvalBatch`.
- [ ] The training and validation steps use dense `forward()` and return scalar losses.
- [ ] Test/prediction use `score_final_query()`, not dense `forward()`.
- [ ] `predict_step()` works when the batch has no `targets`.
- [ ] Prediction output tensors have shape `(B,)`.
- [ ] Scalar evaluation logits match the corresponding dense logits within numerical tolerance in an oracle/unit test.
- [ ] A normal training run works with no QuerySeq evaluation root configured.
- [ ] Requesting test/predict without a QuerySeq root fails clearly.
- [ ] `Trainer.predict()` preserves exact grid/output row alignment.
- [ ] The existing `EQ_predict_multitask` end-to-end output schema and values remain covered by regression tests.
- [ ] Existing conditional-multitask checkpoints remain loadable.
- [ ] During QuerySeq prediction, no tensor of shape `(B,K,V)` is materialized.

## Relevant files

- `src/every_query/data/datamodule.py`
- `src/every_query/data/multitask_dataset.py`
- `src/every_query/data/multitask_eval_dataset.py`
- `src/every_query/model/conditional_multitask_ar_model.py`
- `src/every_query/model/conditional_multitask_lightning.py`
- `src/every_query/predict/predict_multitask.py`
- `src/every_query/predict/configs/predict_multitask.yaml`
- `src/every_query/train/configs/conditional_multitask_ar_config.yaml`
- relevant conditional-multitask dataset/model/CLI tests

## Non-goals

- Changing the conditional multitask architecture or token ordering.
- Adding ontology support to `ConditionalMultitaskARModel`.
- Changing the dense training objective.
- Changing the QuerySeq schema or prediction parquet schema.
- Enabling multi-GPU prediction without an explicit, tested row-order restoration mechanism.
